src/dune_rules/dep_conf_eval.ml

```diff
@@ -57,7 +57,45 @@ let dep_on_alias_rec alias ~loc =
             (Path.Source.to_string_maybe_quoted src_dir)
         ]
 
-let dep expander = function
+let relative d s = Path.build (Path.Build.relative d s)
+
+let expand_include ~expander s =
+  let path = relative (Expander.dir expander) s in
+  let open Memo.Build.O in
+  let+ contents = Build_system.read_file path ~f:Io.read_file in
+  let asts =
+    Dune_lang.Parser.parse_string ~fname:(Path.to_string path)
+      ~mode:Dune_lang.Parser.Mode.Many contents
```

Comment:
The choice of `Many` isn't, but it's inconsistent with how use it in the ordered set lang. I suggest we use `Builld_system.read_sexp` and interpret the list from it.

Reply:
I changed the implementation to use `read_sexp` and it just results in:

```
 Error: too many s-expressions found in input
```

Was your change that we add a `~mode` argument to it for the use case here? I'm gonna go ahead and propose that.

Reply:
Proposal in [4341aa4](https://github.com/ocaml/dune/pull/5442/commits/4341aa460ad5cae401fa4cd06d2c95bc04afdb32)

Reply:
Here's the problem that I'm trying to point.

In this example:
```
(deps (:include foo))
```
The expectation is that `foo` is a `Sexp.t list`.

While in:

```
(flags (:include foo))
```

The expectation is that `foo` is `Sexp.t`.

I found this inconsistence to be jarring.

Reply:
Oh you’re saying maybe the file needs to be a sexp, got it now.

there’s also the `include` vs `:include` difference. I think `(:include ` here would bind a new named variable so I chose `include`

Reply:
> there’s also the include vs :include difference. I think (:include here would bind a new named variable so I chose include

Yes, that's fine. In ordered set lang, we need to distinguish between literals and special forms somehow. That's not necessary for dependency specifications.